src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParseException.cs

```diff
@@ -6,22 +6,141 @@
 namespace System.Text.RegularExpressions
 {
     [Serializable]
-    internal sealed class RegexParseException : ArgumentException
+    public sealed class RegexParseException : ArgumentException
     {
-        private readonly RegexParseError _error; // tests access this via private reflection
-
         /// <summary>Gets the error that happened during parsing.</summary>
-        public RegexParseError Error => _error;
+        public RegexParseError Error { get; }
 
         /// <summary>Gets the offset in the supplied pattern.</summary>
         public int Offset { get; }
 
-        public RegexParseException(RegexParseError error, int offset, string message) : base(message)
+        internal RegexParseException(RegexParseError error, int offset, string message) : base(message)
+        {
+            Error = error;
+            Offset = offset;
+        }
+
+        public RegexParseException(string pattern, RegexParseError error, int offset) : base(MakeMessage(pattern, error, offset))
         {
-            _error = error;
+            Error = error;
```

Comment:
Hmmm, I'm looking at the approved API and it looks like the constructor should only take in `error` and `offset`?

Reply:
Oh I see, you needed the pattern to create the error message.

Reply:
@pgovind, indeed. See my discussion above. The problem here is that each default message expects the main subject of the error: the regex pattern. Hence I added it. See: https://github.com/dotnet/runtime/pull/40902#discussion_r471152282 (I also suggest that maybe we should consider dropping the constructor altogether, the review team wasn't entirely convinced it was needed anyway).

Reply:
The other option is to just render the `pattern` part of each message as an empty string.

Reply:
I made a comment elsewhere. I think the approved, generic constructor can just give a generic message. We won't use it.

Reply:
Ok, then the message should not be dependent on the enum. I'll change this then. That's easier :).

Reply:
@danmosemsft & @pgovind it was partially my misunderstanding. I've resolved this by rolling it back and creating a single, simple message for the ctor that perhaps nobody is going to use anyway ;).

Reply:
Cool, I think this is the right idea for now. We can always open another API proposal to add the constructor that takes in the `string`.